Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/AbstractSqlEamDb.java

```diff
@@ -1894,57 +1895,17 @@ public void processInstanceTable(CorrelationAttribute.Type type, InstanceTableCa
             EamDbUtil.closeConnection(conn);
         }
     }
-    
-        /**
-     * Process the Artifact instance in the EamDb
-     *
-     * @param type EamArtifact.Type to search for
-     * @param instanceTableCallback callback to process the instance
-     * @throws EamDbException
-     */
-    @Override
-    public void processInstanceTableRow(CorrelationAttribute.Type type, int id, InstanceTableCallback instanceTableCallback) throws EamDbException {
-        if (type == null) {
-            throw new EamDbException("Correlation type is null");
-        }
-
-        if (instanceTableCallback == null) {
-            throw new EamDbException("Callback interface is null");
-        }
-
-        Connection conn = connect();
-        PreparedStatement preparedStatement = null;
-        ResultSet resultSet = null;
-        String tableName = EamDbUtil.correlationTypeToInstanceTableName(type);
-        StringBuilder sql = new StringBuilder(3);
-        sql.append("select * from ");
-        sql.append(tableName);
-        sql.append(" WHERE id = ?");
 
-        try {
-            preparedStatement = conn.prepareStatement(sql.toString());
-            preparedStatement.setInt(1, id);
-            resultSet = preparedStatement.executeQuery();
-            instanceTableCallback.process(resultSet);
-        } catch (SQLException ex) {
-            throw new EamDbException("Error getting all artifact instances from instances table", ex);
-        } finally {
-            EamDbUtil.closeStatement(preparedStatement);
-            EamDbUtil.closeResultSet(resultSet);
-            EamDbUtil.closeConnection(conn);
-        }
-    }
-    
     /**
-     * Process the Artifact instance in the EamDb
+     * Process the Artifact instance in the EamDb give a where clause
      *
      * @param type EamArtifact.Type to search for
-     * @param correlationCase  CorrelationCase to filter by
      * @param instanceTableCallback callback to process the instance
+     * @param whereClause query string to execute
      * @throws EamDbException
      */
     @Override
-    public void processCaseInstancesTable(CorrelationAttribute.Type type, CorrelationCase correlationCase, InstanceTableCallback instanceTableCallback) throws EamDbException {
+    public void processInstanceTableWhere(CorrelationAttribute.Type type, String whereClause, InstanceTableCallback instanceTableCallback) throws EamDbException {
```

Comment:
I'd just call this processInstanceTable()

Reply:
There already is a processInstanceTable() which was added with the callback Interface, unless this method should supplant that. processInstanceTable() is a select * from <given table type>.